### DISCUSSION TITLE
values of job_stat

### DISCUSSION THREAD

**Discussion by nim-hrkn** [Tomoki-YAMASHITA/CrySPY#41] [General]
VASPのinterfaceからするとjob_statの値は
done: 'reached required accuracy' がある。入力ファイルで最適化できた。
no_file:  'reached required accuracy' がない。スケジューラーがjobを止めたのかもしれない。
skip: 入力ファイルを作れなかった。
でよろしいですか。
From the VASP interface, the values of job_stat are
done: 'reached required accuracy' is present, or, successfully optimized within the input parameters)
no_file: no 'reached required accuracy'. The job scheduler may stopped the calculation. 
skip: input files could not be created.
Is this correct?

**Comment by Tomoki-YAMASHITA** [comment]
job_statの値はあくまで
job_cryspyの最後の行の`sed -i -e '3 s/^.*$/done/' stat_job`が実行されるかどうかで決まります

done: ジョブスクリプトが最後まで実行された
no_file: ファイルが削除されたり，不測の事態用です．
skip: 手動でskipさせたい時に書き換えた場合

'reached required accuracy'がOUTCARに出力されるかどうかはチェックしていますが，
動作には影響しません．
cryspy_rsltファイルのOptのところに最適化が完了しているかどうかの目安として表示しています．


The value of `job_stat` is only determined by the last line of job_cryspy `sed -i -e '3 s/^. *$/done/' stat_job` .

done: job script was executed to the end
no_file: for file deletion or other contingencies
skip: If you want to skip the job manually, edit the file

 'reached required accuracy' in OUTCAR is checked, but it does not affect the operation.
The "Opt" section of the cryspy_rslt file indicates whether the structure optimization is done or not.

**Comment by yw-fang** [comment]
Thank you very much for presenting the discussions both in Japanese and English!
@nim-hrkn Just a minor append comment: no_file relevant to the VASP jobs generally indicate the absence of the OUTCAR file because the function "check_opt_vasp" will try to open it and examine whether the relaxation calculations in OUTCAR "reached required accuracy". If "reached required accuracy", then "done" value will be retuned to the "check_opt", otherwise "check_opt" is "not_yet". These returned values will be finally implied in the Column of "Opt" of the file data/cryspy_rslt_energy_asc.
Just correct me if I'm wrong @Tomoki-YAMASHITA

**Reply by Tomoki-YAMASHITA** [reply]
Thank you for nice information!
